README.md

```diff
@@ -12,32 +15,59 @@ Thoth is deployed to the following [environments](docs/environments.md).
 
 ## Separation of Application Components and Application Configuration
 
-Each component of the Thoth Application is separated into an ArgoCD Application, there is a core Application
-that contains global configs and secrets.
+Each component of the Thoth Application is separated into an ArgoCD Application.
+The core Application that contains global configs and secrets along with the setup of the namespace based requirements.
 
 ## Policy Based Control of resources
 
-All resources created by kustomize should pass OPA conftest, you could check this by
-using: `kustomize build --enable_alpha_plugins overlays/test | conftest test --policy ../policy -`.
+All resources created by kustomize should pass OPA conftest, you could check this by using: `kustomize build --enable_alpha_plugins overlays/test | conftest test --policy ../policy -`.
 
 ## Use Cases
 
-### testing a pull request
+### Test a pull request
 
-### pushing a release to production
+### Push Release to Production
 
 ### Developing a new component
 
+Developing a new component under thoth-station:
+
+- Please make use of the [template-project](https://github.com/thoth-station/template-project) it has basic requirements. Github Allow users to create new repository with template projects.
+  If your are a new member of Team Thoth and need a repository on Thoth-station. Please create an issue in Thoth-application.
+
+- Once the Developing work is done on the repository, please make use of the Thoth-bots for smooth release. More information on updating the configuration of the Thoth-bots [click here](https://github.com/AICoE/aicoe-ci#setting-aicoe-ci-on-github-organizationrepository).
+
+- Bots also allow user to configure them to auto update the deployment of the component in thoth-application. For this firstly, we have to setup a component for the application.
+  The component must be created based on the description of the project. Following the kustomize methods the component should have two sub-directory **base** and **overlays**, which is required by argocd applications for deployment.
+  Base would contain files which would be deployed across all environments(Prod/Stage/Test), be sure to add all relevant files. (Note: If you feel a file configuration seems like a environment specific thing then it definitely doesn't belong in the base ).
+  Overlays would contains more sub-directories based on environments (Prod/Stage/Test), please follow the pattern of other component. Add the relevant files to the overlays, you are working on.
+
+- Once the component contains all information requirement, we have to add an argo-cd application configuration for the component to be supported by argocd.
```

Comment:
`Once the component contains all information required` 

`argocd` is used in general, or should be argo-cd?